src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -7544,6 +7544,60 @@ public async Task Step(Task<int> t)
             comp.VerifyEmitDiagnostics(expected);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/79415")]
+        public void CompoundAssignment()
+        {
+            var source = """
+                using System;
+                using System.Threading.Tasks;
+
+                interface I1 
+                {
+                    int this[int i] { get; set; }
+                }
+
+                struct S1 : I1
+                {
+                    public int F1;
+
+                    public int this[int i]
+                    {
+                        get
+                        {
+                            Console.Write(F1);
+                            return 0;
+                        }
+                        set
+                        {
+                            Console.Write(F1);
+                        }
+                    }
+                }
+
+                class Program
+                {
+                    static async Task Main()
+                    {
+                        await Test3<S1>();
+                    }
+
+                    static T GetT<T>() where T : I1 => (T)(object)new S1 { F1 = 123 };
+
+                    static async Task Test3<T>() where T : I1
+                    {
+                       GetT<T>()[0] += await Get1Async();
+                    }
+
+                    static async Task<int> Get1Async()
+                    {
+                        await Task.Yield();
+                        return 1;
+                    }
+                }
+                """;
+            CompileAndVerify(source, expectedOutput: "123123").VerifyDiagnostics();
```

Comment:
Please verify runtime async for this test as well, see examples in this file. You're going to conflict with https://github.com/dotnet/roslyn/pull/80093, and depending on who gets in first, one of us is going to need to adjust. I expect there to be an error for this test with runtime async right now; https://github.com/dotnet/roslyn/issues/79763 tracks this, please leave a comment here too.

Please consider moving this test to `CodeGenAsyncSpillTests.cs`, since that's where the rest of the spilling tests are.